timescaledb/tutorials/building-a-custom-timescaledb-dashboard.md

```diff
@@ -0,0 +1,326 @@
+hypertable_name                | conditions
+range_start                    | 2021-05-27 00:00:00+00
+range_end                      | 2021-06-03 00:00:00+00
+before_compression_total_bytes | 90112
+after_compression_total_bytes  | 8192
+```
+
+
+## Setting up your database
+
+In this project, we use data generated by our tutorial on [Simulating IoT sensor data][iot-tutorial]. This data will result in a simple schema and data that mimics a number of IoT sensors with information on time, device, and temperature.
+
+If you follow the tutorial, you will get a table named conditions, which stores the temperature of fake devices over time.
+
+Here’s a set of commands to create the table and generate some sample data:
+
+```sql
+CREATE TABLE conditions (
+      time TIMESTAMPTZ NOT NULL,
+      device INTEGER NOT NULL,
+      temperature FLOAT NOT NULL,
+      PRIMARY KEY(time, device)
+);
+
+SELECT * FROM create_hypertable('conditions', 'time', 'device', 3);
+
+INSERT INTO conditions
+  SELECT time, (random()*30)::int, random()*80 - 40
+  FROM generate_series(TIMESTAMP '2020-01-01 00:00:00',
+                       TIMESTAMP '2020-01-01 00:00:00' + INTERVAL '1 month',
+             INTERVAL '1 min') AS time;
+
+```
+
+
+## Connecting to our database and retrieving metrics
+
+One of the most trivial aspects of a backend application is to protect the database and expose only the information that is accessible to the authorized user. The Hasura GraphQL Engine solves this issue providing GraphQL APIs over new or existing PostgreSQL databases. Allowing you to create permission rules and dynamically expand your database resources.
+
+Now that we have a database example, we can jump into the [Hasura cloud][hasura-cloud] to connect the resources that we want to expose through the GraphQL.
+
+We chose Hasura because of the flexibility to simply connect our TimescaleDB database and quickly expose the tables, views, and functions we needed in a few clicks. We’re not going to dive into all the steps to set up a new data source on Hasura as the wizard will guide you there.
+
+We’re going to use two types of operations:
+
+* Queries and subscriptions: watch a specific query and keep pulling data updates to the client. We’re going to "subscribe" to the chunks metadata’.
+* Mutation: convention for operations that are writing data. We’ll map the compression and decompression actions as mutations.
+
+### Queries and subscriptions
+
+Hasura has great flexibility to plug any resource and offer it as a query or a subscription. For this example, we’ll be mapping the *chunks_with_compression* view we created earlier as a GraphQL resource. It can be consumed as a query or subscription.
+
+![Tracking a SQL view on Hasura cloud](https://assets.timescale.com/docs/images/tutorials/visualizing-compression/hasura-cloud-track-view.png)
+
+We need to map those mutations to compress and decompress a chunk. To accomplish that, we need to take a look at how Hasura types work.
```

Comment:
```suggestion
Hasura allows you to attach any resource and offer it as a query or a subscription. In this example, you map the `chunks_with_compression` view you created earlier as a GraphQL resource, so it can be consumed as a query or subscription. You can then map the changes, or mutations, as you compress and decompress a chunk. This image describes a SQL view is tracked on Hasura:

![Tracking a SQL view on Hasura cloud](https://assets.timescale.com/docs/images/tutorials/visualizing-compression/hasura-cloud-track-view.png)
```